librad/src/git/replication.rs

```diff
@@ -205,8 +205,13 @@ where
                     let rad_id = unsafe_into_urn(
                         Reference::rad_id(Namespace::from(&urn)).with_remote(remote_peer),
                     );
-                    let proj = identities::project::verify(storage, &rad_id)?
-                        .ok_or(Error::MissingIdentity)?;
+                    let proj = identities::project::verify_with(storage, &rad_id, |delegate| {
+                        let refname =
```

Comment:
It is a bit hard to see from the code how the lookup functions are different for the various invocations. Do you think this could be extracted out somehow (maybe curried function "constructors" or something)?